examples/real-dom.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+var document = require('global/document');
+var hg = require('../index.js');
+var h = require('../index.js').h;
+var setTimeout = require('timers').setTimeout;
+
+function Hook() {}
+// Hook to be called when DOM node has been rendered
+Hook.prototype.hook = function hook(node) {
+    node.innerHTML = 'Set directly on real DOM node, by <em>hook</em>.';
+};
+// Hook to be called when DOM node is removed
+Hook.prototype.unhook = function unhook(node) {
+    node.innerHTML = 'Set directly on real DOM node, by <em>unhook</em>.';
+};
+
+function Widget() {
+    this.type = 'Widget';
+}
+
+Widget.prototype.init = function init() {
+    var elem = document.createElement('div');
+    elem.innerHTML = 'Set directly on real DOM node, by widget ' +
```

Comment:
@Raynos I had no idea `innerHTML` was so troublesome. I just saw it used [in the DropzoneJS library](https://github.com/enyo/dropzone/blob/master/src/dropzone.coffee#L1354), which source code I've been studying lately. I don't doubt your judgement here, but would you mind explaining why `innerHTML` is bad, so that I can learn from this?